internal/template/funcmap.go

```diff
@@ -117,7 +119,10 @@ func resourceName(m map[string]interface{}, key string) (string, error) {
 	if !ok {
 		return "", fmt.Errorf("resource name value %v is not a string", v)
 	}
-	return strings.Replace(name, "-", "_", -1), nil
+
+	// invalidIDRE defines the invalid characters not allowed in terraform resource names.
+	invalidIDRE := regexp.MustCompile("[^a-z0-9_]")
```

Comment:
this should be defined in the global scope so its only compiled once and erroor is checked then

Reply:
Done.

Reply:
p.s. 'Must' funcs usually imply they will panic on error which is why you usally want to do them in the global scope. If doing inside a func then call the non-Must version and check error manually

Reply:
Got it. Thanks for the info.